File: src/main/java/com/codecool/repvault/domain/utils/FriendUtil.java
package com.codecool.repvault.domain.utils;

import com.codecool.repvault.domain.entities.User;
import com.codecool.repvault.domain.entities.ids.FriendId;
import com.codecool.repvault.infrastructure.repositories.FriendRepository;
import org.springframework.stereotype.Component;

@Component
public class FriendUtil {
    private final SecurityUtil securityUtil;
    private final FriendRepository friendRepository;

    public FriendUtil(SecurityUtil securityUtil, FriendRepository friendRepository) {
        this.securityUtil = securityUtil;
        this.friendRepository = friendRepository;
    }

    public boolean isFriendsWith(User user) {
        User authenticatedUser = securityUtil.getAuthenticatedUser();
        FriendId id = new FriendId(authenticatedUser.getId(), user.getId());
        return friendRepository.existsById(id);
    }
}

File: src/main/java/com/codecool/repvault/domain/utils/FriendUtil.kt
package com.codecool.repvault.domain.utils

import com.codecool.repvault.domain.entities.User
import com.codecool.repvault.domain.entities.ids.FriendId
import com.codecool.repvault.infrastructure.repositories.FriendRepository
import org.springframework.stereotype.Component

@Component
class FriendUtil(private val securityUtil: SecurityUtil, private val friendRepository: FriendRepository) {
    fun isFriendsWith(user: User): Boolean {
        val authenticatedUser = securityUtil.authenticatedUser
        val id = FriendId(authenticatedUser.id!!, user.id!!)
        return friendRepository.existsById(id)
    }
}
